fs_brokers/apache_hdfs_broker/src/main/java/org/apache/doris/broker/hdfs/FileSystemManager.java

```diff
@@ -268,7 +268,17 @@ public BrokerFileSystem getDistributedFileSystem(String path, Map<String, String
                         long currentTime = System.currentTimeMillis();
                         Random random = new Random(currentTime);
                         int randNumber = random.nextInt(10000);
-                        tmpFilePath = "/tmp/." + Long.toString(currentTime) + "_" + Integer.toString(randNumber);
+                        // different kerberos account has different file
+                        tmpFilePath = "/tmp/." + principal + "_" + Long.toString(currentTime) + "_" + Integer.toString(randNumber);
+                        // before write keytab content into a tmp file, delete if in case of old file confusion
+                        try {
+                            File file = new File(tmpFilePath);
+                            if(!file.delete()){
+                                logger.warn("delete tmp file:" +  tmpFilePath + " failed");
+                            }
+                        } catch (Exception e) {
```

Comment:
Nothing need to warn?

Reply:
I think it is no need. If the same file exists, delete it will sucess. Mostly, the same file will not exist, delete it will failed. It is no need to warn any information. It is a method to make sure it will create a new file at any time.

Reply:
I think it again. It is no need to delete file. Because the file name contains the principal name, kerberos account never be duplicate. Therefore `  tmpFilePath = "/tmp/." + principal + "_" + Long.toString(currentTime) + "_" + Integer.toString(randNumber);` will make the file name unique.